main.nf

```diff
@@ -892,14 +951,14 @@ if (merge) {
 
 // Assemble transcripts
 // ####################
-reference_annotation_to_assemble.combine(maps_to_assemble).set {
-  maps_to_assemble
+reference_annotation_to_assemble.combine(maps_to_merge_for_assembly).set {
```

Comment:
Same comment as above, this should stay `maps_to_assemble` the merging happens before.